pxf/pxf-service/src/main/java/org/apache/hawq/pxf/service/servlet/SecurityServletFilter.java

```diff
@@ -42,8 +52,51 @@
 
     private static final Log LOG = LogFactory.getLog(SecurityServletFilter.class);
     private static final String USER_HEADER = "X-GP-USER";
-    private static final String MISSING_HEADER_ERROR = String.format("Header %s is missing in the request", USER_HEADER);
-    private static final String EMPTY_HEADER_ERROR = String.format("Header %s is empty in the request", USER_HEADER);
+    private static final String SEGMENT_INDEX_HEADER = "X-GP-SEGMENT-ID";
+    private static final String TRANSACTION_ID_HEADER = "X-GP-XID";
+    private static final String MISSING_HEADER_ERROR = "Header %s is missing in the request";
+    private static final String EMPTY_HEADER_ERROR = "Header %s is empty in the request";
+    private static Map<SegmentTransactionId, TimedProxyUGI> cache = new ConcurrentHashMap<>();
+    private static DelayQueue<TimedProxyUGI> delayQueue = new DelayQueue<>();
+    private static long UGI_CACHE_EXPIRY = 15 * 60 * 1000L; // 15 Minutes
```

Comment:
Instead consume this from pxf-site and make it configurable.
{code}
Configuration config = new Configuration();
config.addResource("pxf-site.xml");
{code}

Reply:
I don't think we have pxf-site anymore, config properties are now managed explicitly via pxf-env.sh and are passed as Java properties via -D parameter to Tomcat.

Reply:
We could use pxf-env in that case